### PR TITLE
add sphinx-toggleprompt

### LINKED ISSUE #1095: add toggle for prompts in example code in docs
Currently copying code examples out of the docs is tricky because it's formatted as an IPython notebook.
I propose adding the [sphinx-toggleprompt](https://github.com/jurasofish/sphinx-toggleprompt) extension so these can be toggled.

Similar functionality exists in the offical python docs, numpy docs, and scipy docs (among others)

(nb i am the developer of this library)

### PR DESCRIPTION
**Pull request by jurasofish**
- [x] Closes #1095 
 - [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [x] Tests added
 - [x] Updates entries to [`docs/sphinx/source/api.rst`](https://github.com/pvlib/pvlib-python/blob/master/docs/sphinx/source/api.rst) for API changes.
 - [x] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] New code is fully documented. Includes [numpydoc](https://numpydoc.readthedocs.io/en/latest/format.html) compliant docstrings, examples, and comments where necessary.
 - [x] Pull request is nearly complete and ready for detailed review.
 - [x] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

### CODE REVIEW

**Comment by jurasofish** [comment]
It's not formatting the "Out[19]:" correctly. Will fix

**Comment by jurasofish** [comment]
This is good for review now

File: docs/sphinx/source/whatsnew/v0.8.1.rst [hunk 1]
```diff
@@ -37,6 +37,7 @@ Documentation
 * Update sphinx to 3.1.2 and use the ``recursive`` option in ``autosummary`` class template.
   (:issue:`1055`, :pull:`1075`)
 * Add gallery example about backtracking on sloped terrain. (:pull:`1077`)
+* Add toggle button for code prompts to make copying code easier (:pull:`1096`)
```
Comment (wholmgren) [inline docs/sphinx/source/whatsnew/v0.8.1.rst:40]: add name and/or username to Contributors section below

**Review by wholmgren — APPROVED** [review]
LGTM. Thanks @jurasofish. @pvlib/pvlib-maintainer feel free to merge if you're happy with this approach

**Comment by cwhanse** [comment]
Thanks again @jurasofish

**Comment by kanderso-nrel** [comment]
Looking at the `latest` pages on RTD, it seems like comments might get dropped when hiding the prompts?  For example the [Simple ModelChain example](https://pvlib-python.readthedocs.io/en/latest/modelchain.html#a-simple-modelchain-example).  It seems desirable to me to keep the comments when copy/pasting example code -- is there a way to configure it so comment lines stick around?

**Comment by jurasofish** [comment]
It doesn't hide comments generally, for example see https://docs.python.org/3/tutorial/controlflow.html#for-statements

That modelchain example is a tricky case because the IPython formatting has placed the comment outside an input prompt and formatted it the same as an output, so it's indistinguishable from an output.

The way to tackle it would be to get the upstream IPython formatter to give it a unique style which could then be hidden. Might need changes in pygments, not sure. Or, the IPython formatter could put comments next to "In [xx]:" cells like any normal input.

**Pull request merged**